### ISSUE TITLE
Add unittest to validate TypeScript types

### ISSUE THREAD

**Issue by saschanaz** [mdn/browser-compat-data#16590]
For example, `BrowserNames` is now gone but the schema still refers to it, causing build failure:

https://github.com/mdn/browser-compat-data/blob/2888cf5f5a5a35011acf99e9ae6f037fb6dd50c9/schemas/compat-data.schema.json#L184

There are also several occurrences of `import('../../types').Identifier` in linters which are now all broken:

https://github.com/mdn/browser-compat-data/blob/420c23b0667c34a95d4404507876e21f53a56f18/test/linter/test-status.js#L7

**Comment by saschanaz** [comment]
>This release includes a fundamental change to the TypeScript definitions. 

That sounds like a breaking change 🤔

**Comment by queengooborg** [comment]
TypeScript definitions were not covered by semantic versioning during this release.  However, going forward, they will be (especially since they're auto-generated).  I've just opened up a fix for the typo, and all of those linter typedef imports will be fixed in #16333.

**Comment by saschanaz** [comment]
That's not all. Please reopen and add tests before closing this.

```
node_modules/@mdn/browser-compat-data/types.d.ts:134:44 - error TS2344: Type 'Omit<string, "__compat">' does not satisfy the constraint 'string | number | symbol'.

134 export interface Identifier extends Record<Omit<string, '__compat'>, Identifier> {
                                               ~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@mdn/browser-compat-data/types.d.ts:255:3 - error TS2411: Property '__compat' of type 'CompatStatement | undefined' is not assignable to 'string' index type 'Identifier'.

255   __compat?: CompatStatement;
```
